crates/ty_ide/src/symbols.rs

```diff
@@ -622,4 +1067,837 @@ mod tests {
         assert!(!matches("abcd", "abc"));
         assert!(!matches("δΘπ", "θΔΠ"));
     }
+
+    #[test]
+    fn exports_simple() {
+        insta::assert_snapshot!(
+            public_test("\
+FOO = 1
+foo = 1
+frob: int = 1
+class Foo:
+    BAR = 1
+def quux():
+    baz = 1
+").exports(),
+            @r"
+        FOO :: Constant
+        foo :: Variable
+        frob :: Variable
+        Foo :: Class
+        quux :: Function
+        ",
+        );
+    }
+
+    /// The typing spec says that names beginning with an underscore
+    /// ought to be considered unexported[1]. However, at present, we
+    /// currently include them in completions but rank them lower than
+    /// non-underscore names. So this tests that we return underscore
+    /// names.
+    ///
+    /// [1]: https://typing.python.org/en/latest/spec/distributing.html#library-interface-public-and-private-symbols
+    #[test]
+    fn exports_underscore() {
+        insta::assert_snapshot!(
+            public_test("\
+_foo = 1
+").exports(),
+            @r"
+        _foo :: Variable
+        ",
+        );
+    }
+
+    #[test]
+    fn exports_conditional_true() {
+        insta::assert_snapshot!(
+            public_test("\
+foo = 1
+if True:
+    bar = 1
+").exports(),
+            @r"
+        foo :: Variable
+        bar :: Variable
+        ",
+        );
+    }
+
+    #[test]
+    fn exports_conditional_false() {
+        // FIXME: This shouldn't include `bar`.
```

Comment:
If you're wanting to test fixme cases, another "statically known condition" we should support here is `sys.version_info` checks.